sdk/identity/azure-identity/azure/identity/_internal/msal_credentials.py

```diff
@@ -66,7 +131,12 @@ def _create_app(self, cls):
         # MSAL application initializers use msal.authority to send AAD tenant discovery requests
         with self._adapter:
             # MSAL's "authority" is a URL e.g. https://login.microsoftonline.com/common
-            app = cls(client_id=self._client_id, client_credential=self._client_credential, authority=self._base_url)
+            app = cls(
+                client_id=self._client_id,
+                client_credential=self._client_credential,
+                authority="{}/{}".format(self._authority, self._tenant_id),
+                token_cache=self._cache,
```

Comment:
I don't find the place that we use 'token_cache'?

Reply:
We don't. `cls` here is a subclass of `msal.ClientApplication` i.e. not our code.